osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -420,10 +420,11 @@ private Channel getChannel(Channel lookup, bool addToAvailable = false, bool add
         /// Joins a channel if it has not already been joined. Must be called from the update thread.
         /// </summary>
         /// <param name="channel">The channel to join.</param>
+        /// <param name="switchToJoined">Switch <see cref="CurrentChannel"/> to the joined channel on successful request.</param>
         /// <returns>The joined channel. Note that this may not match the parameter channel as it is a backed object.</returns>
-        public Channel JoinChannel(Channel channel) => joinChannel(channel, true);
+        public Channel JoinChannel(Channel channel, bool switchToJoined = false) => joinChannel(channel, switchToJoined, true);
```

Comment:
I don't like this parameter existing, it doesn't need to exist, and it's also half-misleading (`JoinChannel` will actually switch to the joined channel *anyway* if `CurrentChannel.Value` is null). `ChatOverlay` could just as well set `CurrentChannel.Value` on its own without adding on weird boolean switches to pre-existing methods.

Reply:
The point of this existing here was to switch the channel when it's populated with the messages and all (not immediately switch), but now that I look at it, fetching messages is done in a separate step which may take long before succeding, potentially after `HighlightedMessage` has been set and consumed... probably need to make `DrawableChannel` re-attempt highlighting on new messages rather than just forgetting about it.

Reply:
On re-reading I'm actually intensely confused by whatever it is that `joinChannel()` is doing. It appears to be recursively calling itself in some cases...? Very strange. But yeah as you said this flag you have set there doesn't do anything to protect against the fetch. To be robust you'd have to check for highlight in two places rather than rely on the bindable value change event.